lib/http/options.ts

```diff
@@ -0,0 +1,89 @@
+const __settings__ = {
```

Comment:
Keep this `Settings` only. And better to convert it to `enum`. This is different from `__settings__` inside the class.